media-cdn/snippets.py

```diff
@@ -0,0 +1,217 @@
+            expires=expiration_timestamp,
+            key_name=key_name)
+
+    digest = ed25519.Ed25519PrivateKey.from_private_bytes(
+        decoded_key).sign(policy.encode('utf-8'))
+    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    signature = signature[:-2]  # remove trailing '='
+
+    signed_policy = u'Cloud-CDN-Cookie={policy}:Signature={signature}'.format(
+            policy=policy, signature=signature)
+    return signed_policy
+# [END mediacdn_sign_cookie]
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+            description=__doc__,
+            formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    subparsers = parser.add_subparsers(dest='command')
+
+    sign_url_parser = subparsers.add_parser(
+            'sign-url',
+            help="Sign a URL to grant temporary authorized access.")
+    sign_url_parser.add_argument(
+            'url', help='The URL to sign.')
+    sign_url_parser.add_argument(
+            'key_name',
+            help='Key name for the signing key.')
+    sign_url_parser.add_argument(
+            'base64_key',
+            help='The base64 encoded signing key.')
+    sign_url_parser.add_argument(
+            'expiration_time',
+            type=lambda d: datetime.datetime.utcfromtimestamp(float(d)),
+            help='Expiration time expressed as seconds since the epoch.')
+
+    sign_url_prefix_parser = subparsers.add_parser(
+            'sign-url-prefix',
+            help="Sign a URL prefix to grant temporary authorized access.")
+    sign_url_prefix_parser.add_argument(
+            'url', help='The request URL.')
+    sign_url_prefix_parser.add_argument(
+            'url_prefix', help='The URL prefix to sign.')
+    sign_url_prefix_parser.add_argument(
+            'key_name',
+            help='Key name for the signing key.')
+    sign_url_prefix_parser.add_argument(
+            'base64_key',
+            help='The base64 encoded signing key.')
+    sign_url_prefix_parser.add_argument(
+            'expiration_time',
+            type=lambda d: datetime.datetime.utcfromtimestamp(float(d)),
+            help='Expiration time expessed as seconds since the epoch.')
+
+    sign_cookie_parser = subparsers.add_parser(
+            'sign-cookie',
+            help="Generate a signed cookie to grant temporary authorized access.")
+    sign_cookie_parser.add_argument(
+            'url_prefix', help='The URL prefix to sign.')
+    sign_cookie_parser.add_argument(
+            'key_name',
+            help='Key name for the signing key.')
+    sign_cookie_parser.add_argument(
+            'base64_key',
+            help='The base64 encoded signing key.')
+    sign_cookie_parser.add_argument(
+            'expiration_time',
+            type=lambda d: datetime.datetime.utcfromtimestamp(float(d)),
+            help='Expiration time expressed as seconds since the epoch.')
+
+    args = parser.parse_args()
+
+    if args.command == 'sign-url':
```

Comment:
There is also an "path component" request format in https://cloud.google.com/media-cdn/docs/signed-requests#supported-signing-formats .  Do you plan on documenting that one too?

Reply:
I have not tested using `path component` so not planning to add them now!

Reply:
Isn't the whole point of this code to demonstrate the signed request formats?  Are you planning on adding them later?